### PR TITLE
1.x: Test all private Constructors

### PR DESCRIPTION
**Pull request by vanniktech**

### CODE REVIEW

File: src/main/java/rx/exceptions/Exceptions.java [hunk 1]
```diff
@@ -31,7 +31,7 @@
 
     /** Utility class, no instances. */
     private Exceptions() {
-        throw new IllegalStateException("No instances!");
+        throw new AssertionError("No instances.");
```
Comment (akarnokd) [inline src/main/java/rx/exceptions/Exceptions.java]: Why the change to AssertionError?

Comment (vanniktech) [inline src/main/java/rx/exceptions/Exceptions.java]: I find it that this exception matches the use case a bit better since we're asserting that this constructor shall never be called.

File: src/test/java/rx/EventStreamTest.java [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+package rx;
+
+import com.pushtorefresh.private_constructor_checker.PrivateConstructorChecker;
+
+import org.junit.Test;
+
+public class EventStreamTest {
+    @Test
+    public void constructorShouldBePrivate() {
+        PrivateConstructorChecker.forClass(EventStream.class).expectedTypeOfException(AssertionError.class).expectedExceptionMessage("No instances.").check();
```
Comment (akarnokd) [inline src/test/java/rx/EventStreamTest.java]: These could be refactored into a `TestHelper` enum:

``` java
public enum TestHelper {
    ;
   public static void check(Class<?> clazz) {
        PrivateConstructorChecker.forClass(clazz)
        .expectedTypeOfException(IllegalStateException.class)
        .expectedExceptionMessage("No instances!").check();
    }
}
```

File: src/main/java/rx/exceptions/Exceptions.java (hunk 1 above)
Comment (akarnokd) [inline src/main/java/rx/exceptions/Exceptions.java]: I find these as unnecessary changes. The methods throw to those who instantiate the class via reflection and are not test methods themselves to fail with AssertionError like `TestSubscriber.assertX`.

**Comment by akarnokd** [comment]
:+1: Great!

/cc @artem-zinnatullin

**Comment by artem-zinnatullin** [comment]
👍 

Now I'm not that fanatic regarding testing private constructors, but since test coverage causes addiction and constant desire to increase it — let's merge it!

**Comment by davidmoten** [comment]
Can I suggest in the spirit of DRY that you make a static method available somewhere (a new class `TestingUtil`, say) like this:

``` java
public static void checkConstructorPrivate(Class<?> cls)  {
    PrivateConstructorChecker
         .forClass(cls)
        .expectedTypeOfException(IllegalStateException.class)
        .expectedExceptionMessage("No instances!").check();
}
```

and call that instead?

**Comment by akarnokd** [comment]
Thanks!

@davidmoten I'll refactor those checks into a simple call as you and [I](https://github.com/ReactiveX/RxJava/pull/4125#discussion_r68494047) suggested.
